src/de.rs

```diff
@@ -0,0 +1,491 @@
+use pyo3::type_object::PyTypeInfo;
+use pyo3::types::*;
+use pyo3::{PyObject, Python};
+use serde::{de, Deserialize};
+
+use crate::error::{PythonizeError, Result};
+
+pub fn depythonize<'de, T>(py: Python, obj: PyObject) -> Result<T>
```

Comment:
As `PyObject` is an owned python reference and `&PyAny` is a borrowed one (essentially), I think would be better to take `&'de PyAny` here.

```suggestion
pub fn depythonize<'de, T>(py: Python, obj: &'de PyAny) -> Result<T>
```

Reply:
Sure that does make things simpler